src/app/stores/migrator_baseline.ts

```diff
@@ -0,0 +1,278 @@
+  // TODO are nested charts scopes ?
+
+  return appStoreState;
+}
+
+function upgradeScope(parentElement: Chart, parentState: ChartState) {
+  upgradeChartGuides(parentElement, parentState);
+  upgradeGlyphGuides(parentElement, parentState);
+}
+
+function upgradeChartGuides(parentElement: Chart, parentState: ChartState) {
+  // get chart guides
+  const chartGuideRefs = find(
+    parentElement.elements,
+    parentState.elements,
+    element => element.classID === GuideClass.classID
+  );
+
+  chartGuideRefs.forEach(ref => {
+    const { element, state } = ref;
+
+    // convert mappings to actual values
+    const valueMapping = element.mappings.value as ParentMapping;
+    if (valueMapping && valueMapping.type === "parent") {
+      const { parentAttribute } = valueMapping;
+      // set value to actual mapped attr value
+      state.attributes.value = parentState.attributes[parentAttribute];
+      // remove the mapping
+      delete element.mappings.value;
+    } else {
+      // guides should not be mapped to anything other than parent
+      // Notify user?
+    }
+
+    // find other elements constrained to this chartElementItem
+    parentElement.constraints.forEach(constraint => {
+      if (
+        constraint.type === "snap" &&
+        constraint.attributes.targetElement === element._id
+      ) {
+        changeConstraintTarget(
+          element,
+          constraint,
+          +state.attributes.value,
+          parentElement.elements,
+          parentState.elements
+        );
+      }
+    });
+
+    // add new properties
+    addNewGuideProperties(element, state);
+
+    // remove deleted properties / attributes
+    removeOldGuideProperties(element, state);
+  });
+}
+
+function upgradeGlyphGuides(
+  parentElement: Chart,
+  parentState: ChartState,
+  nested = false
+) {
+  parentElement.glyphs.forEach(glyph => {
+    // collect and separate marks from guides
+    const guides: { [id: string]: Mark } = {};
+    glyph.marks.forEach(mark => {
+      if (isType(mark.classID, GuideClass.classID)) {
+        guides[mark._id] = mark;
+      } else if (isType(mark.classID, NestedChartElementClass.classID)) {
+        const nc = mark as Specification.Element<NestedChartElementProperties>;
+        upgradeGlyphGuides(nc.properties.specification, null, true); // nested charts do not store in ChartState
+      }
+    });
+    // get element which uses this glyph
+    const related = find(
+      parentElement.elements,
+      parentState && parentState.elements,
+      element => {
+        const ps = element as PlotSegment;
+        return ps.glyph === glyph._id;
+      }
+    );
+    // look at constraints
+    glyph.constraints.forEach(constraint => {
+      if (constraint.type === "snap") {
+        const id = constraint.attributes.targetElement as string;
+        const guide = guides[id];
+        if (guide && constraint.attributes.targetAttribute === "value2") {
```

Comment:
Nitpick: "value2" should this be static?

Reply:
converted to enum 👍